library/psa_crypto.c

```diff
@@ -1159,8 +1167,14 @@ psa_status_t psa_destroy_key(mbedtls_svc_key_id_t key)
     /* Unregister from reading the slot. If we are the last active reader
      * then this will wipe the slot. */
     status = psa_unregister_read(slot);
-    /* Prioritize CORRUPTION_DETECTED from unregistering over
-     * a storage error. */
+
+#if defined(MBEDTLS_THREADING_C)
+    PSA_THREADING_CHK_RET(mbedtls_mutex_unlock(
```

Comment:
I'm torn here. The unregister read could be throwing a `CORRUPTION_DETECTED` (and possibly others) and we would potentially be overwriting that here - the storage calls all try to not overwrite errors, should we be doing that here? The lock call is fine, as nothing has happened by then, but this one is more sensitive (and yes I know we don't have a macro for this). Thoughts?

Reply:
The `PSA_THREADING_CHK` macros used here for locking and unlocking don't overwrite `status` unless it equals `PSA_SUCCESS`, the only error which can be overwritten is a storage error in `overall_status` (this was already the case).

Reply:
Ok, misunderstanding, apologies.